### DISCUSSION TITLE
How to handle custom session data?

### DISCUSSION THREAD

**Discussion by realconvis** [shopware/frontends#382] [Q&A] [answered]
I wonder how to handle custom session data.
For example, if I program a Shopware plugin that extends the shop functionality and makes decisions based on the value of a session variable. 

The user is asked for a decision in the shop, for example via modal. 
He makes his selection and the value of this selection should be stored in the session.

Currently, there is no api endpoint for this. 
Should I create one specifically for this? 
But it's not really a separate entity, just a setting option. 
Or would a custom controller be a better choice?

**Comment by BrocksiNet** [comment, marked as answer]
We answered this question in community Slack.

**Quote Björn:**
> Hi
> The question for me is: 
> What happens or is changing for the customer when the cookie or session value is A or B. 
> 
> Also I would think about does it needs to be a session value or if a cookie value is enough.
> 
> Nuxt docus:
> https://nuxt.com/docs/api/composables/use-cookie
> https://nuxt.com/modules/session
> 
> I think the session is more complex to integrate.
> 
> And yes if some endpoint should return something else depending on the session or cookie you need to create it. For frontends, you should always provide store API endpoints and not controllers.
> 
> That's my 50 cents for your question.

**Quote Niklas:**
> In this case I’d go with a custom cookie as Björn suggested. For logged-in customers there are customer groups where exactly this can be configured, but not for not-logged-in users
